simulation.cpp

```diff
@@ -273,7 +273,15 @@ void switch_optimal_function(simulation &s)
 }
 
 size_t get_inds_input_size(const simulation &s)
-{return get_inds_input(s).size();
+
+{
+    return get_inds_input(s).size();
+}
+
+std::function<double(std::vector<double>)> get_current_env_function(const simulation &s)
+{
+    environment e = s.get_env();
```

Comment:
I think the compiler will optimise this away, but to be really sure you could make `e` a 
```suggestion
    const environment& e = s.get_env();
```

Reply:
Or even
```suggestion
    auto e = s.get_env();
```